submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,339 @@
+        this.getHtmlProfile(profile, index)
+      );
+    });
+  }
+
+  getHtmlFilter() {
+    return `   
+      <fieldset class="filter filter-gender">
+        <p>filter by gender</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="male" name="sort-gender">
+        <label for="male" class="radio-lable">Male</label>
+        <input type="radio" id="female" name="sort-gender">
+        <label for="female" class="radio-lable">female</label>
+        <input type="radio" id="allGender" name="sort-gender">
+        <label for="allGender" class="radio-lable">all</label>
+        </div>
+      </fieldset>
+
+      <fieldset class="filter filter-name">
+        <p>sort by name</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="nameAZ" name="sort-name">
+        <label for="nameAZ" class="radio-lable">A - Z</label>
+        <input type="radio" id="nameZA" name="sort-name">
+        <label for="nameZA" class="radio-lable">Z - A</label>
+        <input type="radio" id="nameAll" name="sort-name">
+        <label for="nameAll" class="radio-lable">default</label>
+        </div>
+      </fieldset>
+      
+      <fieldset class="filter filter-age">
+        <p>sort by age</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="age09" name="sort-age">
+        <label for="age09" class="radio-lable">young - old</label>
+        <input type="radio" id="age90" name="sort-age">
+        <label for="age90" class="radio-lable">old - young</label>
+        <input type="radio" id="ageAll" name="sort-age">
+        <label for="ageAll" class="radio-lable">default</label>
+        </div>
+      </fieldset>
+
+      
+      <fieldset class="filter filter-search">
+        <p>Search user by email</p>
+        <input id="nameInput" type="text" size="30" class="searchEmail icon search">
+      </fieldset>
+    `;
+  }
+
+  renderFilters() {
+    this.filtersContainer.insertAdjacentHTML('beforeend', this.getHtmlFilter());
+  }
+}
+
+class Filter {
+  constructor(render) {
+    this.profilesContainer = document.querySelector('.profile-cards');
+    this.renderDataWithFilters = render;
+    render.renderFilters();
+    this.filtersPanel = document.querySelector('#filters');
+    this.searchInput = document.querySelector('#nameInput');
+    this.sortedUsers = [];
+    this.sortByNameTypes = {
+      inAscending: 'nameAZ',
+      inDescending: 'nameZA',
+      byDefault: 'nameAll',
+    };
+    this.sortByAgeTypes = {
+      inAscending: 'age09',
+      inDescending: 'age90',
+      byDefault: 'ageAll',
+    };
+  }
+
+  unckeckedRadioButton(nodeName) {
+    const checkedRadio = document.querySelector(
+      `input[type=radio][name=${nodeName}]:checked `
+    );
+    if (checkedRadio) checkedRadio.checked = false;
+  }
+
+  selectSortMethod(selectedRadioButton) {
+    const name = selectedRadioButton.name;
+    const sortMethod = {
+      'sort-gender': () => {
+        this.unckeckedRadioButton('sort-name');
```

Comment:
Right now this is a magic strings.

Please, put `'sort-gender'` etc into constants and then reuse them.

Also, you should name your arguments in function. ` this.unckeckedRadioButton('sort-age')` does not describes anything.

` this.unckeckedRadioButton({buttonName: 'sort-age' })` does

Reply:
ok, I put all names of sort methods in ```const sortMethods```, and named arguments in function like you said